packages/providers/mgt-electron-provider/src/Provider/ElectronProvider.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { IProvider, Providers, ProviderState, createFromProvider } from '@microsoft/mgt-element';
+import { AuthenticationProviderOptions } from '@microsoft/microsoft-graph-client/lib/es/IAuthenticationProviderOptions';
+import { ipcRenderer } from 'electron';
+
+/**
+ * ElectronProvider class to be instantiated in the renderer process.
+ * Responsible for communicating with ElectronAuthenticator in the main process to acquire tokens
+ *
+ * @export
+ * @class ElectronProvider
+ * @extends {IProvider}
+ */
+export class ElectronProvider extends IProvider {
+  constructor() {
+    super();
+    this.graph = createFromProvider(this);
+    this.setupProvider();
+  }
+
+  /**
+   * Sets up messaging between main and renderer to receive SignedIn/SignedOut state information
+   *
+   * @memberof ElectronProvider
+   */
+  setupProvider() {
+    ipcRenderer.on('isloggedin', async (event, isLoggedIn) => {
```

Comment:
change `isloggedin` to something like `mgtAuthStateChanged` and use an enum instead of a boolean to capture Loading state